examples/xed-asmparse-main.c

```diff
@@ -481,10 +489,46 @@ static void process_operand(xed_enc_line_parsed_t* v,
         operands[i++] = xed_mem_gbisd(seg, base, indx, scale, disp, width_bits);
         process_mem_decorator(q->decorators, operands, &i);
     }
-    else {
+        break;
+    case OPND_FARPTR: {
+        if (*has_imm0) {
+            asp_error_printf(
+                "Long pointer cannot follow immediate operand\n");
+            exit(1);
+        }
+        xed_uint16_t seg = (xed_uint16_t)q->farptr.seg_value;
+        xed_uint32_t offset = (xed_uint32_t)q->farptr.offset_value;
+        xed_uint_t seg_bits = get_constant_width(q->farptr.seg,
+                                                 q->farptr.seg_value);
+        xed_uint_t offset_bits = get_constant_width(q->farptr.offset,
+                                                    q->farptr.offset_value);
+
+        seg_bits = seg_bits < 16 ? 16 : seg_bits;
+        if (seg_bits != 16) {
+            asp_error_printf(
+                "Segment value in far pointer must be 16 bits\n");
+            exit(1);
+        }
+        if (offset_bits > 32) {
```

Comment:
On Intel64, far pointers in 64b mode can have 64b offsets (making for a 2+8 byte payload).